domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/FeatureFlagsLogger.kt

```diff
@@ -0,0 +1,167 @@
+import org.oppia.android.util.platformparameter.EnableLearnerStudyAnalytics
+import org.oppia.android.util.platformparameter.EnableLoggingLearnerStudyIds
+import org.oppia.android.util.platformparameter.EnablePerformanceMetricsCollection
+import org.oppia.android.util.platformparameter.EnableSpotlightUi
+import org.oppia.android.util.platformparameter.FAST_LANGUAGE_SWITCHING_IN_LESSON
+import org.oppia.android.util.platformparameter.INTERACTION_CONFIG_CHANGE_STATE_RETENTION
+import org.oppia.android.util.platformparameter.LEARNER_STUDY_ANALYTICS
+import org.oppia.android.util.platformparameter.LOGGING_LEARNER_STUDY_IDS
+import org.oppia.android.util.platformparameter.PERFORMANCE_METRICS_COLLECTION
+import org.oppia.android.util.platformparameter.PlatformParameterValue
+import org.oppia.android.util.platformparameter.SPOTLIGHT_UI
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Convenience logger for feature flags.
+ *
+ * This logger is meant to be used to log the current status of all feature flags once after the app
+ * has been launched.
+ */
+@Singleton
+class FeatureFlagsLogger @Inject constructor(
+  private val analyticsController: AnalyticsController,
+  @EnableDownloadsSupport
+  private val enableDownloadsSupport: PlatformParameterValue<Boolean>,
+  @EnableExtraTopicTabsUi
+  private val enableExtraTopicTabsUi: PlatformParameterValue<Boolean>,
+  @EnableLearnerStudyAnalytics
+  private val enableLearnerStudyAnalytics: PlatformParameterValue<Boolean>,
+  @EnableFastLanguageSwitchingInLesson
+  private val enableFastLanguageSwitchingInLesson: PlatformParameterValue<Boolean>,
+  @EnableLoggingLearnerStudyIds
+  private val enableLoggingLearnerStudyIds: PlatformParameterValue<Boolean>,
+  @EnableEditAccountsOptionsUi
+  private val enableEditAccountsOptionsUi: PlatformParameterValue<Boolean>,
+  @EnablePerformanceMetricsCollection
+  private val enablePerformanceMetricsCollection: PlatformParameterValue<Boolean>,
+  @EnableSpotlightUi
+  private val enableSpotlightUi: PlatformParameterValue<Boolean>,
+  @EnableInteractionConfigChangeStateRetention
+  private val enableInteractionConfigChangeStateRetention: PlatformParameterValue<Boolean>,
+  @EnableAppAndOsDeprecation
+  private val enableAppAndOsDeprecation: PlatformParameterValue<Boolean>
+) {
+  /**
+   * This method logs the name, enabled status and sync status of all feature flags to Firebase.
+   *
+   * @param sessionId denotes the id of the current appInForeground session
+   */
+  fun logAllFeatureFlags(sessionId: String) {
+    analyticsController.logImportantEvent(
+      createFeatureFlagContext(sessionId = sessionId),
+      null
+    )
+  }
+
+  /**
+   * Collects all the feature flags in the app, creates a FeatureFlagItem for them and adds them to
+   * a list that is returned to the calling object.
+   *
+   * @return a list of [FeatureFlagItem]s
+   */
+  private fun compileFeatureFlagsForLogging(): List<FeatureFlagItem> {
```

Comment:
Could this code be optimized using the map function?

Reply:
PTAL at my implementation of this and make suggestions if necessary.